.github/workflows/sync_branches.yml

```diff
@@ -53,26 +71,38 @@ jobs:
           SYNC_BRANCH: sync-branch-${{ matrix.from_branch}}-to-${{ matrix.to_branch }}
 
       - name: "Merge Changes"
-        if: steps.check_sync.outputs.skip_sync != 'true'
         run: |
-          git checkout -b "${SYNC_BRANCH}"
-          git fetch origin "${TO_BRANCH}"
-          git reset --hard "origin/${TO_BRANCH}"
-          git pull --no-edit origin "${FROM_BRANCH}"
+          echo "::group::Fetching and merging branches"
+          if [ "${BRANCH_EXISTS}" = true ]; then
+            git fetch origin "${SYNC_BRANCH}" "${FROM_BRANCH}"
+            git checkout "${SYNC_BRANCH}"
+          else
+            git fetch origin "${TO_BRANCH}" "${FROM_BRANCH}"
+            git checkout "origin/${TO_BRANCH}"
+            git checkout -b "${SYNC_BRANCH}"
+          fi
+          git config --global user.email "${SYNC_COMMITTER_EMAIL}"
+          git config --global user.name "${SYNC_COMMITTER_NAME}"
+          git pull --no-edit origin "${FROM_BRANCH}" || true
           COMMIT_ORIGINAL="$(git show --no-patch --format="%h" "origin/${TO_BRANCH}")"
-          COMMIT_MERGE="$(git show --no-patch --format="%h" "origin/${TO_BRANCH}")"
+          COMMIT_MERGE="$(git show --no-patch --format="%h" "origin/${SYNC_BRANCH}")"
+          git status
+          echo "::endgroup::"
```

Comment:
What does :: mean?

Reply:
This allow log grouping - [ref](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/workflow-commands-for-github-actions#grouping-log-lines)